.github/workflows/cdk-publish.yml

```diff
@@ -1,8 +1,11 @@
 name: Publish CDK and Source Declarative Manifest
 on:
   push:
+    branches:
+      - main
     paths:
-      - 'airbyte_cdk/pyproject.toml' # To only publish on CDK version change
+      - 'airbyte_cdk/**' # Check for any changes in the CDK directory
+      - '!airbyte_cdk/test/**' # Ignore changes in test CDK test directory
```

Comment:
Since the version in pyproject is static and doesn't reflect the actual dynamic CDK version, we default to pushing a new image on any changes to the CDK (except test files)